### PR TITLE
Implemented feature #18542

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
Add CC request appointment details page styling and interaction.

## Testing done
Unit testing

## Screenshots
Screen Shot 2021-02-08 at 8 16 33 PM

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx [hunk 1]
```diff
@@ -73,6 +77,11 @@ function RequestedAppointmentDetailsPage({
   const typeOfCareText = lowerCase(appointment?.type?.coding?.[0]?.display);
   const facilityId = getVAAppointmentLocationId(appointment);
   const facility = facilityData?.[facilityId];
+  const isCCRequest =
+    appointment.vaos.appointmentType === APPOINTMENT_TYPES.ccRequest;
+  const practitionerName = appointment.contained
+    .filter(res => res.resourceType === 'Practitioner')
+    .map(provider => provider.practitionerRole?.[0].location?.[0].display);
```
Comment (narin) [inline src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx:84]: I think there is a helper called `getPractitionerLocationDisplay()` from the appointment service you can use to get this practitioner name.  If not using that, you probably want to put an an additional optional after the array indexes:

`provider.practitionerRole?.[0]?.location?.[0]?.display`

File: src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx [hunk 2]
```diff
@@ -128,6 +137,16 @@ function RequestedAppointmentDetailsPage({
...
+      {isCC &&
+        isCCRequest && (
+          <>
+            <h2 className="vaos-appts__block-label vads-u-margin-bottom--0 vads-u-margin-top--2">
+              Preferred community care provider
+            </h2>
+            <span>{practitionerName}</span>
+          </>
+        )}
+
```
Comment (narin) [inline src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx:149]: Should we be showing this "Preferred community care provider" header if there is no provider/we have no practitioner name? May want to check with @outerpress

Comment (outerpress) [reply src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx:149]: @narin @vbahinwillit - good catch, there shouldn't be a header if there's no practitioner name.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/components/RequestedAppointmentDetailsPage.jsx:149]: I was wondering about that.

**Review by narin — APPROVED** [review]
LGTM

**Pull request merged**